File: app/laundryapp/model/UserAccount.java
package app.model;

import java.util.List;

public class UserAccount {
    private int userId;
    private String name;
    private String email;
    private String passwordHash;
    private String location;
    private PaymentInfo paymentInfo;
    private List<Order> orders;

    public UserAccount(int userId, String name, String email, String passwordHash, String location,
                       PaymentInfo paymentInfo, List<Order> orders) {
        this.userId = userId;
        this.name = name;
        this.email = email;
        this.passwordHash = passwordHash;
        this.location = location;
        this.paymentInfo = paymentInfo;
        this.orders = orders;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPasswordHash() {
        return passwordHash;
    }

    public void setPasswordHash(String passwordHash) {
        this.passwordHash = passwordHash;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public PaymentInfo getPaymentInfo() {
        return paymentInfo;
    }

    public void setPaymentInfo(PaymentInfo paymentInfo) {
        this.paymentInfo = paymentInfo;
    }

    public List<Order> getOrders() {
        return orders;
    }

    public void setOrders(List<Order> orders) {
        this.orders = orders;
    }

    public void addOrder(Order order) {
        this.orders.add(order);
    }

    @Override
    public String toString() {
        return "UserAccount{" +
                "userId=" + userId +
                ", name='" + name + '\'' +
                ", email='" + email + '\'' +
                ", location='" + location + '\'' +
                ", paymentInfo=" + paymentInfo +
                ", ordersCount=" + (orders != null ? orders.size() : 0) +
                '}';
    }
}

File: app/laundryapp/model/UserAccount.kt
package com.laundryapp.model

class UserAccount {
    // Getters and Setters
    var userId: Int = 0
    var name: String? = null
    var email: String? = null
    var passwordHash: String? = null
    var location: String? = null
    private var paymentInfo: PaymentInfo? = null

    // Constructors
    constructor()

    constructor(
        userId: Int,
        name: String?,
        email: String?,
        passwordHash: String?,
        location: String?
    ) {
        this.userId = userId
        this.name = name
        this.email = email
        this.passwordHash = passwordHash
        this.location = location
    }

    // Method definitions (to be implemented)
    fun register() {}
    fun login() {}
    fun updateProfile() {}
    fun deleteAccount() {}
    fun postJob(job: Job?) {}
    fun rateEmployee(employee: Employee?, rating: Double) {}
    fun requestEmployee(employee: Employee?) {}
    fun blockEmployee(employee: Employee?) {}
    fun bookmarkEmployee(employee: Employee?) {}

    fun getPaymentInfo(): PaymentInfo? {
        return paymentInfo
    }

    fun setPaymentInfo(paymentInfo: PaymentInfo?) {
        this.paymentInfo = paymentInfo
    }
}